src/bundle/DependencyInjection/Compiler/NavigationHubPass.php

```diff
@@ -24,14 +24,23 @@ public function process(ContainerBuilder $container)
 
     private function processNavigationHubTag(ContainerBuilder $container, $tag, $index)
     {
-        $items = [];
+        $unsorted = [];
         $services = $container->findTaggedServiceIds($tag);
-        foreach (array_keys($services) as $serviceId) {
-            $items[] = new Reference($serviceId);
+        foreach ($services as $serviceId => $tags) {
+            foreach ($tags as $tag) {
+                $priority = isset($tag['priority']) ? $tag['priority'] : 0;
+            }
+            $unsorted[$priority][] = new Reference($serviceId);
+        }
+
+        $sorted = [];
```

Comment:
I meant:

```
$container
  ->findDefinition(NavigationHub::class)
  ->replaceArgument(
    $index,
    $this->mapPrioritizedItemsList($container, $services)
  );
```

But again, it's more for readability of the code. Even though what it does isn't that hard to grasp :)